packages/app/src/components/Visualization/StartScreen.js

```diff
@@ -24,20 +24,27 @@ const StartScreen = ({ error }) => {
                 engine,
                 6
             )
-            const visualizations = mostViewedVisualizationsResult.visualization
+            const visualizations = mostViewedVisualizationsResult.visualization // {position: int, views: int, id: string, created: string}
             if (visualizations && visualizations.length) {
                 const visualizationsResult = await apiFetchVisualizations(
                     engine,
-                    visualizations.map(vis => vis.id)
+                    visualizations.map(visualization => visualization.id)
                 )
                 const visualizationsWithType =
-                    visualizationsResult.visualization.visualizations
-                const result = visualizations.map(vis => ({
-                    ...visualizationsWithType.find(
-                        visWithType => visWithType.id === vis.id && visWithType
-                    ),
-                    ...vis,
-                }))
+                    visualizationsResult.visualization.visualizations // {id: string, type: string}
+                const result = []
+
+                visualizations.forEach(visualization => {
```

Comment:
Would it be possible to extract this to a function and add unit tests for it?

Reply:
Sure!

Reply:
Also, I'm wondering if it would be better to turn this around. By only fetching the ids from the `mostViewed` endpoint, you can do this:
```
const mostViewedVisualizationsResult = await apiFetchMostViewedVisualizations(
                engine,
                6
            )
            const mostViewedVisIds =
                mostViewedVisualizationsResult.visualization
            if (mostViewedVisIds && mostViewedVisIds.length) {
                const visualizationsResult = await apiFetchVisualizations(
                    engine,
                    mostViewedVisIds
                )
                const visualizationsWithType =
                    visualizationsResult.visualization.visualizations

                setMostViewedVisualizations(
                    visualizationsWithType.filter(({ id }) =>
                        mostViewedVisIds.includes(id)
                    )
                )
            }
```

Reply:
p.s., I didn't test it yet.

Reply:
I think that would work. But what would the benefit be?